src/webserver.cpp

```diff
@@ -454,13 +458,49 @@ MHD_Result webserver::post_iterator(void *cls, enum MHD_ValueKind kind,
         const char *transfer_encoding, const char *data, uint64_t off, size_t size) {
     // Parameter needed to respect MHD interface, but not needed here.
     std::ignore = kind;
-    std::ignore = filename;
     std::ignore = content_type;
     std::ignore = transfer_encoding;
     std::ignore = off;
 
     struct details::modded_request* mr = (struct details::modded_request*) cls;
-    mr->dhr->set_arg(key, mr->dhr->get_arg(key) + std::string(data, size));
+
+    if (filename == nullptr || mr->ws->file_upload_target != FILE_UPLOAD_DISK_ONLY) {
+        mr->dhr->set_arg(key, mr->dhr->get_arg(key) + std::string(data, size));
+    }
```

Comment:
Would this prevent the library from parsing the other POST parameters whenever the configuation is on?

I guess a unit test would help us see if that's the case.

Reply:
No, it does not (at least if I can trust my tests).

`filename == nullptr` is the indicator, that the currently targeted key in the post_iterator is not an uploaded file. So these parameters will always be put to the arguments.
the `mr->ws->file_upload_target != FILE_UPLOAD_DISK_ONLY` will trigger, that files (where filename is not a nullptr) should be stored to the arguments too.

I did tests with all of the possible file_upload_target values and all parameters, other content (application/json, ...) was all handled as it should.

Yes, a unit test would be helpful. That's on my agenda.

Reply:
I did a few tests on this (with Postman)

The content (to be retrieved with `get_content`) is always correct.

_Upload the same file with different keys_
- Two separate entries in the args map
- Only one file created with doubled content (and only one file in the files map)

_Upload different files with the same key_
- Only one arg with the doubled content
- Two separate files (and two files in the map)

_Upload same file with the same key (this IMHO makes absolutely no sense)_
- Only one arg and one file with duplicated content

This behaviour is actually logical, as the args map is set up by the key and the files map by the filename
Question is probably: What is the expected behavior?

Better solution for the files map would be to set up the map with an unique identifier and always take key and filename into account.

But how can this be changed for the args? As changing the map will change the API and therefor break the compatibility.